code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -525,6 +525,15 @@
 		desc = "Your standard drinking glass."
 		return
 
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/pen))
+		var/txt = stripped_input(user, "What would you like to name this drink?", "Mixology", "", 30)
+		if(txt)
```

Comment:
@bgobandit I'd suggest checking if the player is still alive and next to the `src` when the `if(txt)` runs unless you'd like what @XDTM just referenced to happen ;-)